hooks/generic-webhook/values.yaml

```
@@ -31,3 +31,12 @@ hook:
 
   # hook.tolerations -- Optional tolerations settings that control how the hook job is scheduled (see: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)
   tolerations: []
+
+  # hook.authentication -- Optional basic authentication credentials 
+  authentication:
+    # -- Link a pre-existing generic secret with `usernameKey` and `passwordKey` key / value pairs
+    userSecret: generic-webhook-credentials
+    # -- Name of the username key in the `userSecret` secret. Use this if you already have a secret with different key / value pairs
+    usernameKey: username
+    # -- Name of the password key in the `userSecret` secret. Use this if you already have a secret with different key / value pairs
+    passwordKey: password 
```

I think it might be worth it to add an additional layer here, to clearly mark that the username and password are used for basic auth and also to allow adding other authentication mechanisms (like the auth header) later.

```suggestion
  # hook.authentication -- Optional basic authentication credentials 
  authentication:
    basic:
      # -- Link a pre-existing generic secret with `usernameKey` and `passwordKey` key / value pairs
      userSecret: generic-webhook-credentials
      # -- Name of the username key in the `userSecret` secret. Use this if you already have a secret with different key / value pairs
      usernameKey: username
      # -- Name of the password key in the `userSecret` secret. Use this if you already have a secret with different key / value pairs
      passwordKey: password
```

Note: suggestion can't be applied directly as it also requires a change to the helm template where the vars are used :)

---

Good point. More methods is yet to come:)